### PR TITLE
[Fixes #81] Don't throw on invalid fetcher access

### LINKED ISSUE #81: Exception Fetchr "" could not be found
**Reproduction step***
If you specify your api prefix in a fluxible app with fetchr as: 

```
app.plug(fetchrPlugin({
  xhrPath: "/api"
}));
```

and then you access your api via http:///yourwebsite.com/api, you will get the following exception
<pre>

  File "/home/blahblah/blah/node_modules/fluxible-plugin-fetchr/node_modules/fetchr/libs/fetcher.js", line 78, in Function.Fetcher.getFetcher
    throw new Error('Fetcher "' + name + '" could not be found');
  File "/home/blahblah/blah/node_modules/fluxible-plugin-fetchr/node_modules/fetchr/libs/fetcher.js", line 180, in Function.Fetcher.single
    var fetcher = Fetcher.getFetcher(request.resource.split('.')[0]),
  File "/home/blahblah/blah/node_modules/fluxible-plugin-fetchr/node_modules/fetchr/libs/fetcher.js", line 152, in null.<anonymous>
    Fetcher.single(request);
  File "eval at wrapHandle (/home/blahblah/blah/node_modules/newrelic/lib/instrumentation/express.js", line 1, in eval
  File "[as handle_request] (/home/blahblah/blah/node_modules/express/lib/router/layer.js", line 82, in Layer.handle
  File "/home/blahblah/blah/node_modules/express/lib/router/index.js", line 302, in trim_prefix
    layer.handle_request(req, res, next);
  File "/home/blahblah/blah/node_modules/express/lib/router/index.js", line 270, in null.<anonymous>
    trim_prefix(layer, layerError, layerPath, path);
  File "/home/blahblah/blah/node_modules/express/lib/router/index.js", line 321, in Function.proto.process_params
    return done();
  File "/home/blahblah/blah/node_modules/newrelic/lib/instrumentation/express.js", line 202, in Function.cls_wrapProcessParams
    return process_params.apply(this, arguments)
  File "/home/blahblah/blah/node_modules/express/lib/router/index.js", line 261, in next
    self.process_params(layer, paramcalled, req, res, function (err) {
  File "/home/blahblah/blah/node_modules/newrelic/lib/transaction/tracer/index.js", line 157, in wrapped
    var result = handler.apply(this, arguments)
  File "/home/blahblah/blah/node_modules/csurf/index.js", line 100, in csrf
    next()
  File "/home/blahblah/blah/node_modules/newrelic/lib/transaction/tracer/index.js", line 157, in Server.wrapped
    var result = handler.apply(this, arguments)
  File "/home/blahblah/blah/node_modules/newrelic/lib/instrumentation/core/http.js", line 121, in Server.wrappedHandler
    return tracer.bindFunction(listener, segment).apply(this, arguments)
  File "/home/blahblah/blah/node_modules/newrelic/lib/transaction/tracer/index.js", line 157, in Server.wrapped
    var result = handler.apply(this, arguments)
  File "/home/blahblah/blah/node_modules/newrelic/lib/transaction/tracer/index.js", line 81, in Server.wrapTransactionInvocation
    return tracer.bindFunction(handler, transaction.trace.root).apply(this, arguments)
  File "events.js", line 129, in Server.emit
  File "[as onIncoming] (_http_server.js", line 491, in HTTPParser.parserOnIncoming
  File "_http_common.js", line 111, in HTTPParser.parserOnHeadersComplete
  File "_http_server.js", line 343, in Socket.socketOnData
  File "events.js", line 107, in Socket.emit
  File "_stream_readable.js", line 163, in readableAddChunk
  File "_stream_readable.js", line 126, in Socket.Readable.push
  File "net.js", line 538, in TCP.onread
</pre>

probably needs better handling of errors.

### PR DESCRIPTION
**Pull request by Vijar**
I also fixed some old tests in this PR.

@redonkulus @lingyan @geekyme

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by lingyan** [comment]
:ship:

**Comment by Vijar** [comment]
I'll look into adding some defensive code for POST requests too.

URLs really only matter for GET requests, but for POST we have to validate the post body.

File: libs/fetcher.js [hunk 1]
```diff
@@ -81,6 +82,16 @@ function parseParamValues (params) {
     };
 
     /**
+     * @method hasFetcher
+     * @memberof Fetcher
+     * @param {String} name
+     * @returns {Boolean} true if fetcher with name was registered
+     */
+    Fetcher.hasFetcher = function (name) {
```
Comment (redonkulus) [inline libs/fetcher.js]: Maybe `isRegistered` for method name, a little clearer.

File: libs/fetcher.js [hunk 2]
```diff
@@ -81,6 +82,16 @@ function parseParamValues (params) {
     };
 
     /**
+     * @method hasFetcher
+     * @memberof Fetcher
+     * @param {String} name
```
Comment (redonkulus) [inline libs/fetcher.js]: Add description of what name is

**Comment by redonkulus** [comment]
squash and :+1:

**Pull request merged**